botocore/exceptions.py

```diff
@@ -52,6 +53,16 @@ class EndpointConnectionError(BotoCoreError):
         'Could not connect to the endpoint URL: "{endpoint_url}"')
 
 
+class BadStatusLineError(ConnectionError):
+    fmt = (
+        'Bad status line occurred on endpoint URL: "{endpoint_url}". '
```

Comment:
OK. The tricky thing here is that, the server did not purposely close connection before it send back a proper response; it is just somehow the response (or the connection) was lost during the transmission. So I think we would need some neutral wording.

How about this?

    class ConnectionLost(ConnectionError):
        fmt = 'Connection is lost before we received a valid response from endpoint URL: "{endpoint_url}".'